parabeac.dart

```diff
@@ -1,32 +1,53 @@
 import 'dart:convert';
 import 'dart:io';
+import 'package:args/args.dart';
 
+ArgResults argResults;
+
+// ignore: always_declare_return_types
 main(List<String> args) async {
-  List<String> arguments = ['lib/main.dart'];
-  final helpText = "Usage Options: \n" + "-url \n" + "-key \n" + "-Skey";
-  var url = '';
-  var key = '';
-  var sKey = '';
-  //If arguments is empty or only has -h
-  if (args.length == 0 || args[0] == '-h') {
-    print(helpText);
-  } else {
-    for (var i = 0; i < args.length; i += 2) {
-      switch (args[i]) {
-        case '-url':
-          url = args[i + 1];
-          break;
-        case '-key':
-          key = args[i + 1];
-          break;
-        case '-Skey':
-          sKey = args[i + 1];
-          break;
-        default:
-          arguments.addAll([args[i], args[i + 1]]);
-          break;
-      }
-    }
+  var arguments = <String>['package:parabeac_core/main.dart'];
+
+  //sets up parser
+  //wil set the hide option for help flag to true to
+  //prevent the help usage from being printed twice in console
+  final parser = ArgParser()
+    ..addOption('path',
+        help: 'Path to the design file', valueHelp: 'path', abbr: 'p')
+    ..addOption('out', help: 'The output path', valueHelp: 'path', abbr: 'o')
+    ..addOption('project-name',
+        help: 'The name of the project', abbr: 'n', defaultsTo: 'temp')
+    ..addOption('config-path',
+        help: 'Path of the configuration file',
+        abbr: 'c',
+        defaultsTo: 'lib/configurations/configurations.json')
+    ..addFlag('help',
+        help: 'Displays this help information.', abbr: 'h', negatable: false)
+    ..addOption('url',
+        help: 'S3 bucket link to download and install Parabeac Eggs', abbr: 'u')
+    ..addOption('key', help: 'key for S3 bucket account', abbr: 'k')
+    ..addOption('secret-key', help: 'S3 secret key', abbr: 's');
+
+  argResults = parser.parse(args);
+
+  //Check if no args passed or only -h/--help passed
+  //stops the program after printing the help for both the
+  //main parabeac arguments and the the parabeac egg arguments
+  if (argResults['help'] || argResults.arguments.isEmpty) {
+    print('''
+Common commands:
+
+${parser.usage}
+    ''');
+    exit(0);
+  }
+
+  var url = argResults['url'];
+  var key = argResults['key'];
+  var sKey = argResults['secret-key'];
+
+  if (argResults.arguments.contains('path')) {
```

Comment:
Hey @snwagbata , this seems to not get the arguments passed by the user. For example, if I look into the contents of `argResults.arguments` it contains the following:
`[-p, /home/rootnt/Documents/Parabeac-Core/TestingSketch/parabeac_demo_alt.sketch, -c, default:/home/rootnt/Documents/Parabeac-Core/lib/configurations/configurations.json]`.

However, what is passed to main.dart is only `[package:parabeac_core/main.dart]`.

Reply:
Hey, @ivan-015 , just fixed this in the latest commit